### PR TITLE
support click navigation for breadcrumb

### PR DESCRIPTION
**Pull request by patelvarun**

### CODE REVIEW

File: packages/base/Breadcrumbs/index.js [hunk 1]
```diff
@@ -3,12 +3,13 @@ import React from 'react';
...
 import Button from '../Button';
 
-const Breadcrumbs = ({ crumbs, crumbSx, separator, ...rest }) => (
+const Breadcrumbs = ({ crumbs, crumbSx, navigate, separator, ...rest }) => (
   <Box display="flex" py="4" {...rest}>
     {crumbs.map((crumb, i) => (
       <Button
         key={i}
         link={crumb.path}
+        onClick={navigate ? () => navigate(i) : navigate}
```
Comment (cadejscroggins) [inline packages/base/Breadcrumbs/index.js]: Can we make this more generic? Maybe allow each `crumb` to contain `link` and `onClick` and remove `path`? We'd then pass these props along to `Button`.

Comment (cadejscroggins) [reply packages/base/Breadcrumbs/index.js]: And if neither exist, we'd remove the `cursor: pointer` styles.

File: packages/base/Breadcrumbs/index.js [hunk 2]
```diff
@@ -8,7 +8,8 @@ const Breadcrumbs = ({ crumbs, crumbSx, separator, ...rest }) => (
     {crumbs.map((crumb, i) => (
       <Button
         key={i}
-        link={crumb.path}
+        link={crumb.link}
+        onClick={crumb.onClick ? () => crumb.onClick() : crumb.onClick}
```
Comment (cadejscroggins) [inline packages/base/Breadcrumbs/index.js]: Should just be:

```suggestion
        onClick={crumb.onClick}
```

Comment (patelvarun) [reply packages/base/Breadcrumbs/index.js]: @cadejscroggins  - I passed it as When we have onClick={crumb.onClick} it is giving the below error in the case when we just pass the link.  

TypeError: crumb.onClick is not a function

**Pull request merged**